### PR TITLE
tracing: add feature for analysing post-mortem

### PR DESCRIPTION
**Pull request by mfiumara**
This commit adds a Kconfig option for enabling post-mortem mode of SystemView.
This is useful for debugging crashes that occur after a longer period of time.

Additional PR for the segger module can be found here: https://github.com/zephyrproject-rtos/segger/pull/4

Signed-off-by: Mattia Fiumara <mattia.fiumara@bgrid.com>

### CODE REVIEW

**Comment by chrta** [comment]
You should also update west.yml to point to the pr in the segger repo, see https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-modules

```
    - name: segger
      revision: pull/4/head
      path: modules/debug/segger
```

This way this pr here builds and once zephyrproject-rtos/segger#4 is merged, the revision in this west.yml must be updated to the merged revision.

**Comment by mfiumara** [comment]
@chrta Thanks I was wondering how to compile against a pull request of a module, but that makes sense, will update the PR to reflect this.

**Comment by nashif** [comment]
> @chrta Thanks I was wondering how to compile against a pull request of a module, but that makes sense, will update the PR to reflect this.

would also be great if you can describe and document how you use this option

**Comment by mfiumara** [comment]
Alright thanks for clearing that up, what about Buildkite? Is master not building at all or is it because of this commit?

**Comment by mfiumara** [comment]
@chrta @nashif updated west to point to merged revision in modules/debug/segger.

**Review by chrta — CHANGES_REQUESTED** [review]

File: subsys/tracing/sysview/Kconfig
Comment (mfiumara) [reply subsys/tracing/sysview/Kconfig]: Done

**Review by chrta — APPROVED** [review]
I have not tested this, but LGTM

**Comment by mfiumara** [comment]
Not sure how to add a test for this as it's quite specific. I tested it myself by enabling the option and then using gdb to check if the up buffer is present in RAM: it is then located in `_SEGGER_RTT.aUp[1].pBuffer`.
It can be read out using systemview, or if that doesn't work it can be manually dumped in gdb using:
```
set $start1 = _SEGGER_RTT.aUp[1].pBuffer + _SEGGER_RTT.aUp[1].WrOff
set $end1 = _SEGGER_RTT.aUp[1].pBuffer + _SEGGER_RTT.aUp[1].SizeOfBuffer

set $start2 = _SEGGER_RTT.aUp[1].pBuffer
set $end2 = _SEGGER_RTT.aUp[1].pBuffer + _SEGGER_RTT.aUp[1].RdOff - 1

dump binary memory buf1.bin $start1 $end1
dump binary memory buf2.bin $start2 $end2

dump binary memory sysview_buf.bin $start1 $end1
append binary memory sysview_buf.bin $start2 $end2
```

**Comment by mfiumara** [comment]
@nashif / @wentongwu Could you review this so it can be merged? Thank you for the response.

**Review by wentongwu — APPROVED** [review]

**Comment by nashif** [comment]
@mfiumara can you please rebase and re-push?

**Comment by mfiumara** [comment]
@nashif Done.

File: west.yml
Comment (nashif) [inline west.yml:117]: i think this commit is outdated and has some code that is not compatible with Zephyr anymore. the commit you need is already included in 874d9e9696b00c09f9eeefe839028dc25fe44983, so this change can be removed.

**Review by nashif — CHANGES_REQUESTED** [review]
remove change to west.yml

File: west.yml
Comment (mfiumara) [reply west.yml]: I didn't notice, I have removed this change from the commit.

**Comment by mfiumara** [comment]
Hi @nashif, I'm not sure why I can't get the build to pass but it seems master is broken due to some other errors, correct?

**Review by nashif — APPROVED** [review]

**Pull request merged**
